pkg/solana/chainreader/event_read_binding.go

```diff
@@ -0,0 +1,250 @@
+package chainreader
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/gagliardetto/solana-go"
+
+	commoncodec "github.com/smartcontractkit/chainlink-common/pkg/codec"
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query/primitives"
+
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/codec"
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/logpoller"
+)
+
+type subkeyTuple struct {
+	string
+	uint64
+}
+
+type eventReadBinding struct {
+	namespace, genericName string
+	codec                  types.RemoteCodec
+	modifier               commoncodec.Modifier
+	key                    solana.PublicKey
+	remapper               remapHelper
+	indexedSubkeys         map[string]subkeyTuple
+	events                 EventsReader
```

Comment:
nit: I got confused a few times thinking `events` was a list of events or something, rather than a pointer to logpoller.  Maybe we can name it `reader` or `eventReader`?